### ISSUE TITLE
Electron 0.36 broke stdout on Windows

### ISSUE THREAD

**Issue by Arcanemagus** [atom/atom#11471]
### Prerequisites

* [x] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#check-if-the-problem-shows-up-in-safe-mode)?
* [x] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
* [x] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?
* [x] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
* [x] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)? (I think...)
* [x] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-bug-report) or [enhancement request](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-enhancement-suggestion), see the `CONTRIBUTING` guide.

### Description

It seems that the version of Electron bundled in Atom fails to report any data back on `stdout` when called as Node. `options.env.ELETRON_RUN_AS_NODE` is set (to `1`), but the `stdout` callbacks are never called. If you have Node installed on the machine and replace `process.execPath` with the path to Node everything works as expected (tested with Node v5.9.0).

This has been reported as an issue by users updating to v1.7.0 on the following linter plugins:

* `linter-jshint`
* `linter-htmlhint`
* `linter-csslint`
* Possibly `linter-bootlint` as well (Looks like this might just be user error)

Note that none of those linters have AppVeyor (Windows) CI testing configured.

The following Node based linters _are_ being tested on AppVeyor, but it turns out all of these are directly being `require`'d, and as such aren't being ran as an "external" Node process:

* linter-stylelint
* linter-js-yaml

`linter-eslint` is interesting in that it _is_ working, but it utilizes a worker process that is being created with [`child_process.fork()`](https://github.com/AtomLinter/linter-eslint/blob/v7.2.0/src/helpers.js#L15).

This _might_ be a duplicate of https://github.com/atom/atom/issues/2887, is probably caused by https://github.com/atom/atom/pull/9627, and might be related to https://github.com/atom/atom/pull/11162.

If there is any further information or debugging information that would help please let me know. I'm almost always available as @arcanemagus on Atom's Slack instance if anything is needed.

@50Wliu, @joshaber, @joefitzgerald I'm tagging you three as it seems you have all worked on parts of the Electron update and might know more about how to further test this.

### Steps to Reproduce

1. Open Atom v1.7.0 or above
2. Use something that attempts to load a Node process (ie: BufferedNodeProcess, or more indirectly  [linter-jshint](https://github.com/AtomLinter/linter-jshint))
3. Observe that the `stdout` callbacks are never called

**Expected behavior:** [What you expected to happen]

Calling a Node process works as expected

**Actual behavior:** [What actually happened]

The `stdout` callback is never called, so the process `stdout` never gets any data.

### Versions

OS: Windows 7 - 10
Atom: v1.7.0 - v1.8.0-beta0

Atom v1.8.0-beta0:

```raw
Atom    : 1.8.0-beta0
Electron: 0.36.8
Chrome  : 47.0.2526.110
Node    : 5.1.1
```

**Comment by Arcanemagus** [comment]
Here's a (relatively) small bit of code to reproduce this, place the following as a package in Atom:

`index.js`:
```js
'use babel'

import { CompositeDisposable, BufferedNodeProcess } from 'atom';
import { join } from 'path';

function runExternalNode() {
  const data = { stdout: [] };

  const params = {
    command: join(__dirname, 'node_script.js'),
    stdout(chunk) {
      data.stdout.push(chunk)
    },
    exit() {
      data.stdout = data.stdout.join('').trim();
      console.log(data.stdout);
    }
  };

  let spawnedProcess = new BufferedNodeProcess(params);
}

export default {
  activate() {
    this.subscriptions = new CompositeDisposable();

    this.subscriptions.add(atom.commands.add('atom-text-editor', {
      'node-test:run': () => {
        runExternalNode();
      }
    }));

    runExternalNode();
  },

  deactivate() {
    this.subscriptions.dispose()
  }
}
```

`node_script.js`:
```js
console.log('42');
```

`package.json`:
```json
{
  "private": true,
  "name": "node-test",
  "description": "Test for running Node scripts within Atom",
  "version": "0.0.1"
}
```

For convenience here is a ZIP of those files: [node-test.zip](https://github.com/atom/atom/files/219903/node-test.zip)

----

When launching Atom with that package I get a blank line logged to the console, a breakpoint placed in the stdout() callback is never triggered. If you replace `process.execPath` with the path to a local Node install and then call the `Node Test: Run` command then `42` is output to the console as expected.

Note that we have tested this with the `child_process` module with the same results.

**Comment by BinaryMuse** [comment]
Thanks, @Arcanemagus, that's super helpful

**Comment by Arcanemagus** [comment]
Updated the description with clearer notes on a few packages that are failing, and a few that are _not_ and how those are ran.

**Comment by lee-dohm** [comment]
Fixed in v1.7.1.

**Comment by benoitcerrina** [comment]
Thanks guys for a quick resolution.
